3 - PB/Documentazione esterna/Piano di Qualifica/Piano di Qualifica.typ

```diff
@@ -246,65 +246,165 @@ Dati:
+#figure(
+   table(
+      columns: 3,
+      rows: (auto, 30pt),
+      [*Calcolo della metrica*],[*Valore ottimale*],[*Valore accettabile*],
+      align(center+horizon, [*Ease of Use*]), align(center+horizon,$<=5$), align(center+horizon,$<=7$),
+    ),
+    caption: "Ease of Use"
+)
+
+
+== Manutenibilità
+=== CC (Ciclomatic Complexity)
+La CC è una metrica utilizzata per misurare la complessità di un metodo. Essa fornisce una stima della complessità strutturale del codice sorgente contando il numero di cammini linearmente indipendenti attraverso il grafo di controllo del flusso del metodo.
+#figure(
+   table(
+      columns: 3,
+      rows: (auto, 30pt),
+      [*Calcolo della metrica*],[*Valore ottimale*],[*Valore accettabile*],
+      align(center+horizon, [*CC*$(G) = e - n + p$]), align(center+horizon,$<=7$), align(center+horizon,$<=10$),
+    ),
+    caption: "Ciclomatic Complexity"
+
+)
+Nella formula  _G_ indica il grafo del controllo di flusso, _e_ il numero di archi di _G_, _n_ il numero di nodi di _G_ e _p_ il numero di componenti connesse da ogni arco.
+
+=== CL (Coupling Level)
+Il CL misura il grado di dipendenza di una classe da altre classi nel sistema. Questa dipendenza può manifestarsi in vari modi, come l'invocazione di metodi di altre classi, il riferimento a istanze di altre classi, o la dipendenza da tipi definiti in altre classi.
+#figure(
+   table(
+      columns: 3,
+      rows: (auto, 30pt),
+      [*Calcolo della metrica*],[*Valore ottimale*],[*Valore accettabile*],
+      align(center+horizon, [*Coupling Level*]), align(center+horizon,$<=4$), align(center+horizon,$<=6$),
+    ),
+    caption: "Coupling Level"
+)
+
+=== RC (Responsability Count)
+L'RC misura il numero di responsabilità che una classe ha all'interno di un sistema software. Il concetto di responsabilità seguito è quello illustrato da Robert C. Martin, detto _Uncle Bob_, nel libro _Clean Code: A Handbook of Agile Software Craftsmanship_.
+#figure(
+   table(
+      columns: 3,
+      rows: (auto, 30pt),
+      [*Calcolo della metrica*],[*Valore ottimale*],[*Valore accettabile*],
+      align(center+horizon, [*Responsability Count*]), align(center+horizon,$1$), align(center+horizon,$1$),
+    ),
+    caption: "Responsability Count"
+)
+
+=== MPN (Method Parameters Number)
+Il MPN è una metrica che misura il numero di parametri di un metodo.
+#figure(
+   table(
+      columns: 3,
+      rows: (auto, 30pt),
+      [*Calcolo della metrica*],[*Valore ottimale*],[*Valore accettabile*],
+      align(center+horizon, [*Method Parameters Number*]), align(center+horizon,$<=3$), align(center+horizon,$<=4$),
+    ),
+    caption: "Method Parameters Number"
+)
+
+== Affidabilità
+=== FD (Failure Density)
+La FD è un indicatore della stabilità e della qualità del software. Questa metrica misura il numero di errori o difetti rilevati nel software rispetto alla dimensione o alla complessità del sistema.
+#figure(
```

Comment:
Ho compreso il senso della metrica, ma la trasformerei in un "test passati/test totali" che probabilmente è di più facile comprensione rispetto a valori <= 0.01 (che non credo siano significativi imo)

Reply:
La metrica è molto comune in statistica e di solito si basa sui test falliti in rapporto a quelli eseguiti. Ho cambiato i valori abbassandoli a 0% (quindi ci si aspetta che il nostro mvp passi tutti i test) perchè forse sarebbe più professionale mantenere la metrica definita in questo modo. Dimmi se va bene così o se la vuoi esprimere in funzione dei test passati (anche se in questo caso cambierà nome perchè non è più una densità degli errori)

Reply:
Ok ok va bene. Probabilmente vedere quel 0.01% mi aveva parzialmente disorientato sull'obiettivo della metrica. Alla fine test passati/test totali è complementare a dire test falliti/test totali.

Quindi va bene.